mplc/scenario.py

```diff
@@ -801,55 +770,66 @@ def to_dataframe(self):
             dict_results["contributivity_scores"] = contrib.contributivity_scores
             dict_results["contributivity_stds"] = contrib.scores_std
             dict_results["computation_time_sec"] = contrib.computation_time_sec
-            dict_results[
-                "first_characteristic_calls_count"
-            ] = contrib.first_charac_fct_calls_count
+            dict_results["first_characteristic_calls_count"] = contrib.first_charac_fct_calls_count
 
             for i in range(self.partners_count):
                 # Partner-specific data
                 dict_results["partner_id"] = i
-                dict_results["dataset_fraction_of_partner"] = self.amounts_per_partner[
-                    i
-                ]
+                dict_results["dataset_fraction_of_partner"] = self.amounts_per_partner[i]
                 dict_results["contributivity_score"] = contrib.contributivity_scores[i]
                 dict_results["contributivity_std"] = contrib.scores_std[i]
 
                 df = df.append(dict_results, ignore_index=True)
 
         return df
 
-    def run(self):
+    def run(self, is_dry_run=False):
+
+        # -----------------
+        # Preliminary steps
+        # -----------------
+
+        if not is_dry_run:
+            logger.info("Now starting running a scenario")
+            self.log_scenario_description()
+            self.initialize_output_attributes()
+
         # -----------------------
-        #  Provision the scenario
+        # Provision the scenario
         # -----------------------
 
         self.instantiate_scenario_partners()
+
         # Split data according to scenario and then pre-process successively...
         # ... train data, early stopping validation data, test data
         if self.samples_split_type == "basic":
             self.split_data()
         elif self.samples_split_type == "advanced":
             self.split_data_advanced()
-        self.plot_data_distribution()
+
+        if not is_dry_run:
+            self.plot_data_distribution()
         self.compute_batch_sizes()
         self.data_corruption()
```

Comment:
Why don't move these functions to the .__init__ ? 
However the use of the is_dry_run great may be, these methods look like initialization to me ?

Reply:
That is indeed a good point.
We should probably keep the plot out of it, as a separate method available in the SDK for users, but not in the init. Unless it includes saving the plot in the results folder, in which case it makes sense to have it generated (but not displayed by the python IDE / env).

Yes you are right, it doesn't bring any benefit to keep it out. Maybe only the posibility to execute custom ways to do these things. But that won't happen often...

Reply:
Yes I am not sure for the handling of the plot.
For the other initialization step, it would be great to have them all in the init. Otherwise, a mpl needs a scenario which have been ran, which is strange. 
For instance, when working with a notebook on a new mpl method, I need a scenario to initialize it, and I need to run it to initialize the partners.